infer/environment.cc

```diff
@@ -1164,7 +1164,8 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
                                                            core::UntypedMode::AlwaysCompatible)) {
                     if (auto e = ctx.beginError(bind.loc, core::errors::Infer::ReturnTypeMismatch)) {
                         auto ownerData = ctx.owner.data(ctx);
-                        e.setHeader("Returning value that does not conform to method result type");
+                        e.setHeader("Expected `{}` but found `{}` for method result type", methodReturnType.show(ctx),
```

Comment:
```suggestion
                        e.setHeader("Expected return type to be `{}` but found `{}`", methodReturnType.show(ctx),
```

Reply:
I was planning to factor this all out into a helper in a future PR, where the form is always “Expected X but found Y for Z” (this is already the form that argument type errors and others fall into).

@Morriar how strongly do you feel here?

Reply:
Not strongly at all, but when reading the error, I'd prefer to have the "*return* type" information first so I know what I need to think of before reading the expected type.

Reply:
Sounds good. I’ll leave it as is for now, and in the PR to make all the errors more consistent I’ll revisit it.